### PR TITLE
Check for proper mimetype in tags-that-block audit.

### PR DESCRIPTION
**Pull request by ebidel**
R: @patrickhulce all

Resources that fail or are missing a mimetype are now filtered out of the blocking link/scripts gatherer. This also changes the dbw_tester to not wrap these resources in a `<template>` so their parser initiated and not script initiated.

See #1431 for a full explanation.

### CODE REVIEW

**Comment by patrickhulce** [comment]
Technically wouldn't a script or stylesheet that 404s still block first paint until the failed request came back as 404? Probably not a big deal over h2, but maybe still worth reporting?

Thanks for fixing! Funny that the DBW test scripts were never actually blocking

**Comment by ebidel** [comment]
Yea, true. Let me remove the 404 check.

**Comment by ebidel** [comment]
PTAL. Added a comment on the `?delay` param.

**Comment by ebidel** [comment]
Results are correct:

screen shot 2017-01-09 at 10 15 48 am

**Comment by brendankenny** [comment]
Should we still report on the 404 assets? Waiting for the 404 response could still take a while in a mobile environment.

(plus it's not bad to warn about 404ed assets; OTOH maybe it's more appropriate for a separate 404 audit or something)

**Comment by ebidel** [comment]
> Should we still report on the 404 assets?

We can (probably should for completeness), but what do you think the behavior should be? `record._mimeType == undefined` in those cases, so we can't proceed with the rest of the new "is parser generated" checks the audit does.

I was also thinking a separate 404/"wasted resources" audit would be useful.

**Comment by patrickhulce** [comment]
> record._mimeType == undefined in those cases, so we can't proceed with the rest of the new "is parser generated" checks the audit does.

Is the initiator information still there? We could still link it up with the tags later on and shift the logic down there (for parser initiator)

**Comment by ebidel** [comment]
PTAL. Like pokemon, we're collecting all the 404s now:

screen shot 2017-01-09 at 2 32 29 pm

File: lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js [hunk 1]
```diff
@@ -91,19 +91,25 @@ function collectTagsThatBlockFirstPaint() {
 
 function filteredAndIndexedByUrl(networkRecords) {
   return networkRecords.reduce((prev, record) => {
-    // Filter stylesheet, javascript, and html import mimetypes.
-    const isHtml = record._mimeType.indexOf('html') > -1;
+console.log(record._initiator.type, record._mimeType, record._url)
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js]: remove?

Comment (ebidel) [inline lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js]: haha yep.

Comment (ebidel) [inline lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js]: done

**Comment by ebidel** [comment]
PTAL

**Comment by patrickhulce** [comment]
LGTM

**Pull request merged**